### PR TITLE
Iteration two

### PR DESCRIPTION
**Pull request by reid-andrew**
Add methods/test coverage for all methods in Iteration Two.

### CODE REVIEW

File: test/stat_tracker_test.rb [hunk 1]
```diff
@@ -38,4 +38,73 @@ def test_it_stores_a_collection_of_game_teams_data
...
+    }
+
+    assert_equal expected, @stat_tracker.count_of_games_by_season
+  end
+
+  def test_it_calculates_average_goals_per_game
+    assert_equal 4.27, @stat_tracker.average_goals_per_game
+  end
+
+  def test_it_calculates_total_goals_per_season
```
Comment (reid-andrew) [inline test/stat_tracker_test.rb:82]: This method isn't called for in the Interaction pattern. I split it out as a helper method for average goals per season to keep overall method length at an acceptable level.

File: test/stat_tracker_test.rb [hunk 2]
```diff
@@ -38,4 +38,73 @@ def test_it_stores_a_collection_of_game_teams_data
...
+  def test_it_counts_games_by_season
+    expected = {
+      '20122013' => 51,
+      '20132014' => 82,
+      '20142015' => 85,
+      '20152016' => 87,
+      '20162017' => 82,
+      '20172018' => 84
+    }
+
```
Comment (tylerpporter) [inline test/stat_tracker_test.rb:74]: Let's remove these empty spaces above the assertions.

File: lib/game.rb [hunk 3]
```diff
@@ -37,4 +37,13 @@ def initialize(params)
     @venue_link = params[:venue_link]
   end
 
+  def total_score
```
Comment (reid-andrew) [inline lib/game.rb:40]: These two aren't specifically in interaction pattern. Using them as helper for methods at Stat Tracker.

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -21,4 +21,70 @@ def initialize(games_data, teams_data, game_teams_data)
...
+  def count_of_games_by_season
+    Game.all.reduce({}) do |games_count, game|
+      if !games_count[game.season.to_s]
+        games_count[game.season.to_s] = 1
+      else
+        games_count[game.season.to_s] += 1
+      end
+      games_count
+    end
+  end
```
Comment (tylerpporter) [inline lib/stat_tracker.rb:63]: Great method here! Very readable.

**Pull request merged**